elks/fs/minix/dir.c

```diff
@@ -89,11 +89,11 @@ static int minix_readdir(struct inode *inode,
 	do {
 	    de = (struct minix_dir_entry *) (offset + bh->b_data);
 	    if (de->inode) {
-		if (filldir(dirent, de->name, strnlen(de->name, info->s_namelen),
-			filp->f_pos, de->inode) < 0) {
+			filldir(dirent, de->name, strnlen(de->name, info->s_namelen),
+				filp->f_pos, de->inode);
+			filp->f_pos += info->s_dirsize;
 		    unmap_brelse(bh);
```

Comment:
So we're unmapping even if filldir() returns < 0?

Reply:
The callback function `filldir()` is now called once, so after the call, the job is done, we can properly unmap and return, no more need to test the return value.